### DISCUSSION TITLE
Spaceship login failed, when running Fastlane match

### DISCUSSION THREAD

**Discussion by crazymanish** [fastlane/fastlane#16812] [Q&A] [answered]
Hi @joshdholtz , Good morning!
Today, with the latest Fastlane 2.151.2 version, we faced a very strange problem
**bundle exec fastlane match adhoc**

In my machine, match is working fine 💪🏻, but my team-mate is facing an issue, In his machine, match is asking for the Apple portal password. Please note, a couple of weeks back, everything was fine in his machine. Now facing the issue, so wondering if is it a regression due to recent SpaceShip changes? 🤷🏻‍♂️

match-repo git clone was fine for him, you can see the difference in the attached images, Can you let me know, what's happening? how can I debug this issue? :p 
Thank you so much! 🙏🏻

<img width="850" alt="Screenshot 2020-07-09 at 20 44 03" src="https://user-images.githubusercontent.com/5364500/87080295-b6867680-c227-11ea-9f72-9e0a7a7ef6df.png">

**Comment by evschu** [comment]
Not sure if related, but I'm having an issue with my Apple account that has two factor authentication enabled after upgrading to 2.151.2. We recently enabled 2FA, and added the `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD` environment setting. I believe it was working up till 2.150.3. But now it is asking for a phone number so we can't have an unattended session.

![image](https://user-images.githubusercontent.com/5094734/87084718-89b07000-c1e3-11ea-9b1f-4f168e9119bd.png)

From environment file:

![image](https://user-images.githubusercontent.com/5094734/87084911-de53eb00-c1e3-11ea-8c4b-58f1a7f232e7.png)

**Reply by joshdholtz** [reply]
The session should be good for 30ish days when you auth with a phone number. Are you doing this locally or on a CI?

**Reply by evschu** [reply]
Locally, but eventually will do on CI. Sounds like we can get away with just entering the code for now. Just wanted to point out because this seemed to work in previous versions.

**Reply by joshdholtz** [reply]
The enforcement of 2FA is putting a damper on things. But good news... I’m fairly certain that we will soon be able to use match, cert, sigh, produce, deliver, and produce all with the JWT auth 😇 

This will prevent all the 2FA stuff. I got a lot of work to do for that but that will really be what we all need/want 😊

**Reply by evschu** [reply]
Many thanks for your help. I'll keep an eye out for updates on the 2FA issues.

**Comment by joshdholtz** [comment, marked as answer]
@crazymanish Sometimes the App Store Connect API session maybe have no longer have been valid? The sessions don’t last forever. I like they are like 30ish days? If you teammate enter the password once, does it ask a second time when its run again?

Also... could it be possible that your teammate deleted the entry in his keychain?

**Reply by crazymanish** [reply]
> @crazymanish Sometimes the App Store Connect API session maybe have no longer have been valid? 

That I notice, He don't have any cookie under .fastlane/spaceship/user/cookie file

> The sessions don’t last forever. I like they are like 30ish days? If you teammate enter the password once, does it ask a second time when its run again?

Problem is, we have one common Apple Developer account and we usually don't share that password with all the team-mates, my next question is there no other way, match was not asking password earlier :p

**Reply by joshdholtz** [reply]
Waaaaaiiiiittttt... if you are using `--readonly` it shouldn’t be verifying things on the dev portal 🤔 Did it always do that verifying when you provide `--readonly`?

**Reply by crazymanish** [reply]
> Also... could it be possible that your teammate deleted the entry in his keychain?

He informed me that he cleaned-up some certificates from keychain, will that break match? btw what you mean **deleted the entry in his keychain** you mean password? 🤔  @joshdholtz

**Reply by crazymanish** [reply]
> Waaaaaiiiiittttt... if you are using `--readonly` it shouldn’t be verifying things on the dev portal 🤔 Did it always do that verifying when you provide `--readonly`?

--readonly works fine:) No issue there, no password required:)

**Reply by joshdholtz** [reply]
@crazymanish Yup yup! _fastlane_ stores the password for that in Keychain Access. My guess is either the password from Keychain Access or the session expire and/or he changed the password so the entry maybe have gotten deleted? 

If he enters the password and then it works a second time I’m guessing it was one of the above ☝️ 

If match keeps requiring the password every time (after the first time) then I think we got a 🐛 

<hr/>

Can you let me know if keeps asking on subsequent runs?

**Reply by crazymanish** [reply]
Cool then, I will check that tomorrow with him, Good night! (from my timezone, its 10pm in Amsterdam):)
I will let you know the status tomorrow, thanks again! 🙌🏻

and Thank you so much for creating/setting-up the GitHub discussion for Fastlane. ❤️

**Reply by joshdholtz** [reply]
Awesome! I will be on the lookout for your response 😊 

And not a problem! I’m excited to see how this helps clean up the issues section 😇

**Reply by crazymanish** [reply]
Hello @joshdholtz 
My team-mate deleted the AppStoreConnect password entry (`deliver.email`) from the keychain and that's why Spaceship login was failing for him. Thank you so much for your help! 🙏🏻
